modules/passkey/test/userStories/rotatePasskeyOwner.ts

```diff
@@ -0,0 +1,243 @@
+
+    return {
+      user,
+      proxyFactory,
+      safeModuleSetup,
+      module,
+      entryPoint,
+      singleton,
+      signerFactory,
+      navigator,
+      verifier,
+      SafeL2,
+      signer,
+      credential,
+    }
+  })
+
+  it('should execute a userOp with replaced WebAuthn signer as Safe owner', async () => {
+    // Step 1: Setup the contracts
+    const { user, proxyFactory, safeModuleSetup, module, entryPoint, verifier, singleton, navigator, SafeL2, signer, signerFactory } =
+      await setupTests()
+
+    // Step 2: Create a userOp, sign it with EOA wallet
+
+    // The initializer data to enable the Safe4337Module as a module on a Safe
+    const initializer = safeModuleSetup.interface.encodeFunctionData('enableModules', [[module.target]])
+
+    // Create setup data to deploy a Safe with EOA and passkey signer as owners, threshold 1, Safe4337Module as module and fallback handler
+    const setupData = singleton.interface.encodeFunctionData('setup', [
+      [user.address, signer],
+      1n,
+      safeModuleSetup.target,
+      initializer,
+      module.target,
+      ethers.ZeroAddress,
+      0,
+      ethers.ZeroAddress,
+    ])
+
+    // Predict the Safe address to construct the userOp, generate
+    const safeSalt = Date.now()
+    const safe = await proxyFactory.createProxyWithNonce.staticCall(singleton, setupData, safeSalt)
+
+    // Deploy data required in the initCode of the userOp
+    const deployData = proxyFactory.interface.encodeFunctionData('createProxyWithNonce', [singleton.target, setupData, safeSalt])
+
+    const safeOp = buildSafeUserOpTransaction(
+      safe,
+      ethers.ZeroAddress,
+      0n,
+      '0x',
+      await entryPoint.getNonce(safe, 0),
+      await entryPoint.getAddress(),
+      false,
+      true,
+      {
+        initCode: ethers.solidityPacked(['address', 'bytes'], [proxyFactory.target, deployData]),
+      },
+    )
+
+    const packedUserOp = buildPackedUserOperationFromSafeUserOperation({
+      safeOp,
+      signature: '0x',
+    })
+
+    // Build the signature that a Safe can decode and verify
+    const signature = buildSignatureBytes([
+      {
+        signer: signer as string,
+        data: buildSignatureBytes([await signSafeOp(user, await module.getAddress(), safeOp, await chainId())]),
+      },
+    ])
+
+    // Send 1 ETH to the Safe
+    await user.sendTransaction({ to: safe, value: ethers.parseEther('1') }).then((tx) => tx.wait())
+    // Check if Safe is not already created
+    expect(await ethers.provider.getCode(safe)).to.equal('0x')
+
+    // Step 3: Execute the userOp that deploys a safe with EOA and passkey signer as owners
+    await (
+      await entryPoint.handleOps(
+        [
+          {
+            ...packedUserOp,
+            signature: ethers.solidityPacked(['uint48', 'uint48', 'bytes'], [safeOp.validAfter, safeOp.validUntil, signature]),
+          },
+        ],
+        user.address,
+      )
+    ).wait()
```

Comment:
Nit, I think the Safe setup:
1. Doesn't have to be over 4337 (just adds noise IMO). I would just call the Safe proxy factory directly.
2. Can be in the `setupTests` fixture (as it is the pre-req to the user story and not part of it)

Reply:
https://github.com/safe-global/safe-modules/pull/343/commits/a4b74b962d5b6f95a934fcf819b936161809bfe3